c/driver/postgresql/copy/writer.h

```diff
@@ -105,9 +105,7 @@ class PostgresCopyFieldWriter {
 class PostgresCopyFieldTupleWriter : public PostgresCopyFieldWriter {
  public:
   void AppendChild(std::unique_ptr<PostgresCopyFieldWriter> child) {
-    int64_t child_i = static_cast<int64_t>(children_.size());
```

Comment:
The current design makes it so that the array_view is limited to being supplied to the writers through this `AppendChild` function; however, to support arbitrarily nested writers I think it makes more sense to be moved outside of this

The flip side is now there are a  lot of `writer.Init()` calls placed in the factory function. I also tried to just make the ArrayView part of the constructor for each writer, but that became kind of strange when trying to create the root writer itself.

Probably a better way to structure/refactor this, but decided to live with the code duplication for now to focus on the rest of implementation